### PR TITLE
refactor: cleanup createSiteAddon function

### PR DESCRIPTION
**Pull request by erezrokah**
Related to https://github.com/netlify/cli/issues/310 and https://github.com/netlify/cli/issues/1331

Wanted to fix https://github.com/netlify/cli/issues/1331, but couldn't do it since the addons code doesn't allow getting the env variables easily at the moment:

I going to do some refactoring first so we can have https://github.com/netlify/cli/blob/b88cf6db2028741b93ce271cfaca934c4f812a0d/src/utils/dev.js#L25 return addons envs instead of mutating `process.env`

I removed a bunch of outdated comments from `utils/addons.js` and did some code cleanup.
Note that part of that function is duplicated from:
https://github.com/netlify/cli/blob/b88cf6db2028741b93ce271cfaca934c4f812a0d/src/commands/addons/create.js#L155

I follow up in another PR. 

Also changed some code in `functions/create.js` to use `async/await` instead of promise nesting and added a missing `await`

The code itself is used when running `netlify functions:create` and will add missing addons depending on the function template.
For example if you install a `fauna` template it adds the `fauna` addon

### CODE REVIEW

File: src/commands/functions/create.js [hunk 1]
```diff
@@ -346,36 +346,33 @@ function installAddons(addons = [], fnPath) {
...
-                if (confirmPostInstall) addonDidInstall(fnPath)
-              }
+    const siteData = await api.getSite({ siteId })
+    const arr = addons.map(async ({ addonName, addonDidInstall }) => {
+      this.log(`${NETLIFYDEVLOG} installing addon: ` + chalk.yellow.inverse(addonName))
+      try {
+        const addonCreated = await createSiteAddon(api.accessToken, addonName, siteId, siteData, this.log)
+        if (addonCreated) {
+          if (addonDidInstall) {
+            const { addEnvVariables } = require('../../utils/dev')
```
Comment (ehmicky) [inline src/commands/functions/create.js:356]: Do we have a good reason to dynamically load `require()` here (I think not)?

Comment (erezrokah) [reply src/commands/functions/create.js:356]: Definitely not

File: src/utils/addons.js [hunk 2]
```diff
@@ -1,311 +1,42 @@
...
-    // const deleteCmd = chalk.cyan(`\`netlify addons:delete ${addonName}\``)
-    // log(`- To remove this add-on run: ${deleteCmd}`)
-    // log()
     return false
   }
 
-  // const manifest = await getAddonManifest(addonName, accessToken);
-
-  const configValues = rawFlags
-  // if (manifest.config) {
```
Comment (ehmicky) [inline src/utils/addons.js:33]: :broom:

File: src/utils/addons.js [hunk 3]
```diff
@@ -1,311 +1,42 @@
-/* eslint no-console: 0 */
 const { getAddons, createAddon } = require('netlify/src/addons')
-// const chalk = require("chalk");
-// const fetch = require("node-fetch");
 
-/** main section - shamelessly adapted from CLI. we can extract and dedupe later. */
-/** but we can DRY things up later. */
```
Comment (ehmicky) [inline src/utils/addons.js:7]: :)

**Review by ehmicky — APPROVED** [review]

**Pull request merged**